heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManagerMetrics.java

```diff
@@ -0,0 +1,181 @@
+
+  private final JVMMetrics jvmMetrics;
+  private final MultiCountMetric executeSensorCount;
+  private final MultiCountMetric executeDetectorCount;
+  private final MultiCountMetric executeDiagnoserCount;
+  private final MultiCountMetric executeResolverCount;
+
+  private NIOLooper looper;
+  private HeronClient metricsMgrClient;
+  private Communicator<Metrics.MetricPublisherPublishMessage> outMetricsQueues;
+
+  /**
+   * constructor to expose healthmgr metrics to local metricsmgr
+   * @param metricsMgrPort local MetricsMgr port
+   * @throws IOException
+   */
+  public HealthManagerMetrics(int metricsMgrPort) throws IOException {
+    jvmMetrics = new JVMMetrics();
+
+    executeSensorCount = new MultiCountMetric();
+    executeDetectorCount = new MultiCountMetric();
+    executeDiagnoserCount = new MultiCountMetric();
+    executeResolverCount = new MultiCountMetric();
+
+    looper = new NIOLooper();
+
+    SystemConfig systemConfig =
+        (SystemConfig) SingletonRegistry.INSTANCE.getSingleton(SystemConfig.HERON_SYSTEM_CONFIG);
+
+    HeronSocketOptions socketOptions =
+        new HeronSocketOptions(systemConfig.getInstanceNetworkWriteBatchSize(),
+            systemConfig.getInstanceNetworkWriteBatchTime(),
+            systemConfig.getInstanceNetworkReadBatchSize(),
+            systemConfig.getInstanceNetworkReadBatchTime(),
+            systemConfig.getInstanceNetworkOptionsSocketSendBufferSize(),
+            systemConfig.getInstanceNetworkOptionsSocketReceivedBufferSize(),
+            systemConfig.getInstanceNetworkOptionsMaximumPacketSize());
+    metricsMgrClient =
+        new MetricsMgrClient(looper, METRICS_MGR_HOST, metricsMgrPort, socketOptions);
+
+    outMetricsQueues = new Communicator<Metrics.MetricPublisherPublishMessage>(null, looper);
+    outMetricsQueues.init(systemConfig.getInstanceInternalMetricsWriteQueueCapacity(),
+        systemConfig.getInstanceTuningExpectedMetricsWriteQueueSize(),
+        systemConfig.getInstanceTuningCurrentSampleWeight());
+
+    int interval = (int) systemConfig.getHeronMetricsExportInterval().getSeconds();
+
+    looper.registerTimerEvent(Duration.ofSeconds(interval), new Runnable() {
+      @Override
+      public void run() {
+        jvmMetrics.getJVMSampleRunnable().run();
+
+        // push to container 0 metricsMgr
+        if (!metricsMgrClient.isConnected()) {
+          return;
+        }
+
+        LOG.info("Flushing all pending data in MetricsManagerClient");
+        // Collect all tuples in queue
+        int size = outMetricsQueues.size();
+        for (int i = 0; i < size; i++) {
+          Metrics.MetricPublisherPublishMessage m = outMetricsQueues.poll();
+          metricsMgrClient.sendMessage(m);
+        }
+      }
+    });
+  }
+
+  public Communicator<Metrics.MetricPublisherPublishMessage> getMetricsQueue() {
+    return outMetricsQueues;
+  }
+
+  public synchronized void executeSensorIncr(String sensor) {
```

Comment:
Curious why you not using java-> java.util.concurrent.atomic.AtomicLong package, it is thread safe with better performance.

synchronized with the method which makes the code single thread only even there are multiple threads running.

Reply:
I just wanted to reuse the existing CountMetric, however CountMetric does not have atomic interface